app/stac_api/pgtriggers.py

```diff
@@ -119,12 +127,32 @@ class UpdateCollectionSummariesTrigger(pgtrigger.Trigger):
         RETURN asset_instance;
         '''
 
-        def get_declare(self, model):
-            return [
-                ('asset_instance', 'stac_api_asset%ROWTYPE'),
-                ('related_collection_id', 'INT'),
-                ('collection_summaries', 'RECORD'),
-            ]
+    class UpdateItemUpdateIntervalTrigger(pgtrigger.Trigger):
+        when = pgtrigger.After
+        declare = [
+            ('asset_instance', 'stac_api_asset%ROWTYPE'),
+            ('item_update_interval', 'RECORD'),
+        ]
+        func = '''
+        asset_instance = COALESCE(NEW, OLD);
+
+        -- Compute item update_interval (minimum aggregation of asset's update_interval)
+        SELECT
+            COALESCE(MIN(NULLIF(asset.update_interval, -1)), -1) AS min_update_interval
```

Comment:
Not sure if I understand this correctly, but won't this return -1 as min, in case -1 is present? Do we want this? -1 means the data is not updated on a regular basis. Don't we want to have the minimum value of those items, that ARE updated on a regular basis?

Reply:
`MIN(NULLIF(asset.update_interval, -1))` => returns either a positive value (including `0`) or `null`. It returns `null` if all `update_interval` are `-1`
`COALESCE(.., -1)` return `-1` if the first part is `null`.
We want the minimal positive value (including 0) or -1 if there is no minimal value.

Reply:
OK, thanks, now I got it :wink: